File: 0001-two-sum/0001-two-sum.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int i,j;
    
    *returnSize=2;
    int* a=(int*)malloc(*returnSize*sizeof(int));
    for(i=0;i<numsSize;i++)
        for(j=i+1;j<numsSize;j++)
            if(nums[i]+nums[j]==target){
                a[0]=i;
                a[1]=j;
                return a;
            }
    return NULL;
                
    
}

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
      HashMap<Integer,Integer> mmp=new HashMap();
      for(int i=0;i<nums.length;i++)
      {
            int moreneeded=target-nums[i];
            if(mmp.containsKey(moreneeded))
                return new int[]{mmp.get(moreneeded),i};
            else
                mmp.put(nums[i],i);
      }
      return new int[]{-1,-1};  
    }
}